lib/sync/mongodbQueue.js

```diff
@@ -34,10 +34,12 @@ function MongodbQueue(name, metrics, opts) {
   }
   this.queueName = name;
   this.metrics = metrics;
+  this.lock = lock;
+  this.lockName = opts.lockName || ('lock:sync:' + this.queueName);
+  this.lockTimeout = opts.lockTimeout || 10000;
```

Comment:
is 10 seconds enough for the task to finish when there are a lot of dataset clients?
I think there were numbers in excess of 10 seconds during load testing.